modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/BlockSearchRequest.scala

```diff
@@ -0,0 +1,12 @@
+package org.tessellation.rosetta.server.model
+
+case class BlockSearchRequest(
+  isOr: Boolean,
+  isAnd: Boolean,
+  addressOpt: Option[String],
+  networkStatus: Option[String],
+  limit: Option[Long],
+  offset: Option[Long],
+  transactionHash: Option[String],
```

Comment:
what is the purpose of this field?

Reply:
This allows for the hash of a specific transaction in a block to be specified in the request call.

https://www.rosetta-api.org/docs/BlockApi.html#blocktransaction

Reply:
but it's a search api (https://www.rosetta-api.org/docs/SearchApi.html) and a block search request, not transaction search request.

Reply:
It is used in the [BlockApi](https://www.rosetta-api.org/docs/BlockApi.html).

Reply:
Yes, so it's used in a different request than you model here (it's used in block api, not in the search api)

Reply:
additionally, in the API you provided above, there is transaction hash (identifier) required to get given transaction from a specific block (you specify the block identifier and transaction identifier you want to get), so following REST API specification, it's a request to the `/block/transaction` endpoint.
Here, you have `BlockSearchRequest`, so you search not a transaction but a block. The example provided by rosetta is to search transaction, that's why you need to provide a transaction identifier. When you change transaction->block, you should provide a block identifier to search for.

Reply:
I misspoke. `BlockSearchRequest` is used by the [`/search/transaction` endpoint](https://www.rosetta-api.org/docs/SearchApi.html#searchtransactions). The `/block` endpoints do not use intermediary request objects. One endpoint does use the `BlockResponse` intermediary response object. I have renamed `BlockSearchRequest` to `SearchTransactionRequest`.